packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -99,15 +99,49 @@ def processable_bet(self, bet: Bet) -> bool:
 
     def _sampled_bet_idx(self, bets: List[Bet]) -> int:
         """
-        Sample a bet and return its id.
-
-        The sampling logic is relatively simple at the moment.
-        It simply selects the unprocessed bet with the largest liquidity.
+        Sample a bet and return its index.
+
+        The sampling logic follows the specified priority logic:
+        1. Filter out all the bets that have a processed_timestamp != 0 to get a list of new bets.
+        2. If the list of new bets is not empty:
+           2.1 Order the list in decreasing order of liquidity (highest liquidity first).
+           2.2 For bets with the same liquidity, order them in decreasing order of market closing time (openingTimestamp).
+        3. If the list of new bets is empty:
+           3.1 Order the bets in decreasing order of invested_amount.
+           3.2 For bets with the same invested_amount, order them in increasing order of processed_timestamp (least recently processed first).
+           3.3 For bets with the same invested_amount and processed_timestamp, order them in decreasing order of liquidity.
+           3.4 For bets with the same invested_amount, processed_timestamp, and liquidity, order them in decreasing order of market closing time (openingTimestamp).
```

Comment:
Why increasing order of processed timestamp and why decreasing order of market closing time? Sorry if this already mentioned in the spec.

Reply:
This is laid out in the specs doc in the 'What should be the ordering of the markets–Rational' section

Reply:
From the spec:

> The mech is likely to give out a more well-informed response 
    a. true for older markets
    b. Amongst the older markets–more likely to happen for markets that have been processed less recently

Therefore, I think that a. translates to *increasing* order of market creation - not closing - time.

To be clarified -> maybe the spec meant "true for markets closing sooner", therefore, *increasing* order of market closing time.

Reply:
> More formally: The set of all available markets should be ordered as follows: 
> 
> Decreasing order of investment amount (highest investment first)
> Increasing order of last_processed_time (least recently processed first)
> Decreasing order of liquidity (highest liquidity first)
> Decreasing order of market_closing_time (added for completeness, as a tie-breaker when the above are all same)


@Adamantios  I have followed this final laid plan by Juli in the doc, will confirm with here